internal/util/crypto.go

```diff
@@ -80,6 +80,68 @@ func FetchEncryptionKMSID(encrypted, kmsID string) (string, error) {
 	return kmsID, nil
 }
 
+type EncryptionType int
+
+const (
+	// EncryptionTypeInvalid signals invalid or unsupported configuration.
+	EncryptionTypeInvalid EncryptionType = iota
+	// EncryptionTypeNone disables encryption.
+	EncryptionTypeNone = iota
+	// EncryptionTypeBlock enables block encryption.
+	EncryptionTypeBlock = iota
+	// EncryptionTypeBlock enables file encryption (fscrypt).
+	EncryptionTypeFile = iota
+)
+
+const (
+	EncryptionTypeBlockString = "block"
+	EncryptionTypeFileString  = "file"
+)
+
+func ParseEncryptionType(typeStr string) EncryptionType {
+	switch typeStr {
+	case EncryptionTypeBlockString:
+		return EncryptionTypeBlock
+	case EncryptionTypeFileString:
+		return EncryptionTypeFile
+	case "":
+		return EncryptionTypeNone
+	default:
+		return EncryptionTypeInvalid
+	}
+}
+
+func EncryptionTypeString(encType EncryptionType) string {
+	switch encType {
+	case EncryptionTypeBlock:
+		return EncryptionTypeBlockString
+	case EncryptionTypeFile:
+		return EncryptionTypeFileString
+	case EncryptionTypeNone:
+		return ""
+	case EncryptionTypeInvalid:
+		return "INVALID"
+	default:
+		return "UNKNOWN"
```

Comment:
define this as typed const and return it?

Reply:
The string representation is only used when printing the EncryptionType type. Using a typed const would mean that we need a lot of conversions back to string across the codebase. All checks are done on the EncryptionType. The reason EncryptionTypeBlockString and EncryptionTypeFileString exist is a linter error about the repeated use of what appears to be string constants. Looking at them they are better not exported.